hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableTokenRelationStore.java

```diff
@@ -69,16 +69,15 @@ public void put(@NonNull final TokenRelation tokenRelation) {
      * @param accountId - the number of the account to be retrieved
      * @param tokenId   - the number of the token to be retrieved
      */
-    @NonNull
-    public Optional<TokenRelation> getForModify(@NonNull final AccountID accountId, @NonNull final TokenID tokenId) {
+    @Nullable
+    public TokenRelation getForModify(@NonNull final AccountID accountId, @NonNull final TokenID tokenId) {
         requireNonNull(accountId);
         requireNonNull(tokenId);
 
-        if (AccountID.DEFAULT.equals(accountId) || TokenID.DEFAULT.equals(tokenId)) return Optional.empty();
+        if (AccountID.DEFAULT.equals(accountId) || TokenID.DEFAULT.equals(tokenId)) return null;
 
-        final var token = Objects.requireNonNull(tokenRelState)
+        return Objects.requireNonNull(tokenRelState)
```

Comment:
can we do that check directly in the constructor?

Reply:
Done.